src/sql/workbench/contrib/connection/browser/connectionStatus.ts

```diff
@@ -59,16 +64,38 @@ export class ConnectionStatusbarItem extends Disposable implements IWorkbenchCon
 	// Update the connection status shown in the bar
 	private _updateStatus(): void {
 		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		let id = undefined;
+		if (this._editorService.activeEditor) {
+			// USE ACTIVE EDITOR INFO AS THE ID WILL BE DIFFERENT FOR EDITOR CONNECTION.
+			let newInfo = this._connectionManagementService.getConnectionInfo(this._editorService.activeEditor.resource.toString());
+			id = newInfo?.serverConnectionId;
+		}
 		if (activeConnection) {
-			this._setConnectionText(activeConnection);
+			this._setConnectionText(activeConnection, id)
 			this.show();
-		} else {
+		}
+		else {
 			this.hide();
 		}
 	}
 
+	private _refreshIDStatus(uri: string, id: string | undefined): void {
+		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		if (this._editorService.activeEditor) {
+			// USE ACTIVE EDITOR INFO AS THE ID WILL BE DIFFERENT FOR EDITOR CONNECTION.
+			let currUri = this._editorService.activeEditor.resource.toString();
+			let info = this._connectionManagementService.getConnectionInfo(currUri);
+			if (currUri === uri) {
+				if (info) {
+					info.serverConnectionId = id;
+					this._setConnectionText(activeConnection, id);
```

Comment:
So what happens if I :

1. Start a long-running query
2. Switch to a different tab
3. The query finishes - but since a different editor is active it doesn't update the status (expected)
4. Switch to the original tab

Will the SPID be correct at that point? I don't see it being stored anywhere, which will be necessary to handle that scenario if I'm understanding this all correctly.

Reply:
Fixed so the SPID will be updated for the connection information correctly even if it's not the focused one.